backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -173,40 +192,27 @@ public Endpoint createEndpoint(@Context SecurityContext sec, @NotNull @Valid End
             throw new BadRequestException("Properties is required");
         }
 
-        if (obEnabled) {
-            // TODO NOTIF-429 - see similar in EndpointResources#createEndpoint
-            String endpointSubType;
-            if (endpoint.getSubType() != null) {
-                endpointSubType = endpoint.getSubType();
-            } else {
-                if (endpoint.getType() == EndpointType.CAMEL) {
-                    endpointSubType = endpoint.getProperties(CamelProperties.class).getSubType();
-                } else {
-                    endpointSubType = "not-defined"; // No Camel endpoint, so we can skip
-                }
-            }
+        if (isForOpenBridge(endpoint)) {
+
+            try {
+                String bridgeId = bridge.getId();
+                String token = bridgeAuth.getToken();
+                CamelProperties cp = endpoint.getProperties(CamelProperties.class);
 
-            if (endpointSubType != null && endpointSubType.equals("slack")) {
-                CamelProperties properties = endpoint.getProperties(CamelProperties.class);
                 String processorName = "p-" + endpoint.getAccountId() + "-" + UUID.randomUUID();
-                properties.getExtras().put(OB_PROCESSOR_NAME, processorName);
-                String processorId = null;
-                try {
-                    processorId = setupOpenBridgeProcessor(endpoint, properties, processorName);
-                    endpoint.setStatus(EndpointStatus.PROVISIONING);
-                } catch (Exception e) {
-                    Log.warn("Processor setup failed: " + e.getMessage());
-                    endpoint.setStatus(EndpointStatus.FAILED);
-                    throw new InternalServerErrorException("Can't set up the endpoint");
-                }
+                cp.getExtras().put(OB_PROCESSOR_NAME, processorName);
 
-                // TODO find a better place for these, that should not be
-                //       visible to users / OB actions
-                //       See also CamelTypeProcessor#callOpenBridge
-                //            and ReadyCheck#execute
-                properties.getExtras().put(OB_PROCESSOR_ID, processorId);
+                UnaryOperator<Processor> execFun = in -> bridgeApiService.addProcessor(bridgeId, token, in);
+                UnaryOperator<Processor> updateFun = in -> { endpoint.setStatus(EndpointStatus.PROVISIONING); return in; };
+                Processor out = executeObAction(endpoint, updateFun, execFun);
```

Comment:
The code would be easier to understand/maintain if it looked like this:
```suggestion
                endpoint.setStatus(EndpointStatus.PROVISIONING);
                Processor out = executeObAction(endpoint, updateFun, execFun); // TODO Handle null updateFun in executeObAction
```